hoot-core/src/main/cpp/hoot/core/visitors/IndexElementsVisitor.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "IndexElementsVisitor.h"
+
+// Hoot
+#include <hoot/core/Factory.h>
+#include <hoot/core/OsmMap.h>
+#include <hoot/core/index/OsmMapIndex.h>
+#include <hoot/core/conflate/NodeToWayMap.h>
+#include <hoot/core/ops/RecursiveElementRemover.h>
+#include <hoot/core/util/ConfigOptions.h>
+
+// TGS
+#include <tgs/RStarTree/IntersectionIterator.h>
+
+namespace hoot
+{
+
+IndexElementsVisitor::IndexElementsVisitor(shared_ptr<HilbertRTree>& index,
+                                           deque<ElementId>& indexToEid,
+                                           const shared_ptr<ElementCriterion>& filter,
+                                           boost::function<Meters (const ConstElementPtr& e)> getSearchRadius,
+                                           ConstOsmMapPtr pMap):
+  _pMap(pMap),
+  _filter(filter),
+  _getSearchRadius(getSearchRadius),
+  _index(index),
+  _indexToEid(indexToEid)
+{
+  // This space intentionally left blank
+}
+
+void IndexElementsVisitor::setOsmMap(OsmMap* map)
+{
+  // Do nothing with this map. We want to conform to the interface,
+  // but prefer to use the ptr supplied in the constructor
+  (void) map;
```

Comment:
I was reviewing my changes last night, and thinking this looked particularly suspect... I think this is in there because IndexElementVisitor is inheriting from something that it doesn't need.